apps/meteor/server/models/raw/LivechatDepartment.ts

```diff
@@ -151,4 +188,115 @@ export class LivechatDepartmentRaw extends BaseRaw<ILivechatDepartmentRecord> im
 	removeDepartmentFromForwardListById(_departmentId: string): Promise<void> {
 		throw new Error('Method not implemented in Community Edition.');
 	}
+
+	async saveDepartmentsByAgent(agent: { _id: string; username: string }, departments: string[] = []): Promise<void> {
+		const { _id: agentId, username } = agent;
+		const savedDepartments = (await LivechatDepartmentAgents.findByAgentId(agentId).toArray()).map((d) => d.departmentId);
+
+		const incNumAgents = (_id: string, numAgents: number) => this.updateOne({ _id }, { $inc: { numAgents } });
+		// remove other departments
+		const deps = difference(savedDepartments, departments).map(async (departmentId) => {
+			// Migrate func
+			await LivechatDepartmentAgents.removeByDepartmentIdAndAgentId(departmentId, agentId);
+			await incNumAgents(departmentId, -1);
+		});
+
+		await Promise.all(deps);
+
+		const promises = departments.map(async (departmentId) => {
+			const dep = await this.findOneById(departmentId, {
+				projection: { enabled: 1 },
+			});
+			if (!dep) {
+				return;
+			}
+
+			const { enabled: departmentEnabled } = dep;
+			// Migrate func
+			const saveResult = await LivechatDepartmentAgents.saveAgent({
+				agentId,
+				departmentId,
+				username,
+				departmentEnabled,
+				count: 0,
+				order: 0,
+			});
+
+			if (saveResult.upsertedId) {
+				incNumAgents(departmentId, 1);
+			}
+		});
+
+		await Promise.all(promises);
+	}
+
+	updateById(_id: string, update: Partial<ILivechatDepartment>): Promise<Document | UpdateResult> {
+		return this.updateOne({ _id }, update);
+	}
+
+	updateNumAgentsById(_id: string, numAgents: number): Promise<Document | UpdateResult> {
+		return this.updateOne({ _id }, { $set: { numAgents } });
+	}
+
+	findEnabledWithAgents(projection: FindOptions<ILivechatDepartment>['projection'] = {}): FindCursor<ILivechatDepartment> {
+		const query = {
+			numAgents: { $gt: 0 },
+			enabled: true,
+		};
+		return this.find(query, projection && { projection });
+	}
+
+	findEnabledWithAgentsAndBusinessUnit(
```

Comment:
I see an override for this function over here within base model which adds some custom login of Business unit. So wanted to confirm if that wouldn't get affected by this migration

Reply:
Not yet, as this only migrates the function. There's another PR for updating usages